app/models/principal.rb

```diff
@@ -66,7 +68,7 @@ def associate_firm
   end
 
   def match_fca_number
-    unless Lookup::Firm.exists?(fca_number: self.fca_number)
+    unless self.fca_number.blank? || Lookup::Firm.exists?(fca_number: self.fca_number)
```

Comment:
I thought the same and thought I'd tried it and failed.  Will give it another shot.